rir/src/compiler/util/visitor.h

```diff
@@ -88,13 +112,26 @@ class VisitorImplementation {
      *
      */
     static void run(BB* bb, InstrAction action) {
+        static_assert(DIR == Direction::Forward,
+                      "Backward visitors need a CFG.");
         run(bb, [action](BB* bb) {
             for (auto i : *bb)
                 action(i);
         });
     }
 
+    static void run(CFG const& cfg, BB* bb, InstrAction action) {
+        static_assert(DIR == Direction::Backward,
+                      "Only backward visitors can take a CFG.");
```

Comment:
would it maybe make sense to use a subclass for the backwards visitor? Then you can add the cfg to the constructor.

Reply:
Hmm maybe I could factor out the `scheduleNext` and then the subclass would work I think.. I will give it another go

Reply:
ah, right, I see

Reply:
i guess it's fine like that in that case. don't worry